geowebcache/core/src/main/java/org/geowebcache/storage/blobstore/file/FileBlobStore.java

```diff
@@ -673,16 +767,15 @@ public void putLayerMetadata(final String layerName, final String key, final Str
         }
     }
 
-    private Properties getLayerMetadata(final String layerName) {
-        final File metadataFile = getMetadataFile(layerName);
+    private Properties getUncompressedLayerMetadata(final File metadataFile) {
```

Comment:
Multiple thread/processes could be trying to migrate the old property file too. Suggest using optimistic checks here too, also consider that during migration the file can disappear, if the other "actor" in place manages to do the migration before we do.

Reply:
The position of the comment was probably a bit confusing. If someone else manages to do the migration before this code does, and the file disappears, it should check the compressed metata file.